pkg/alertmanager/operator.go

```diff
@@ -804,7 +806,10 @@ func (c *Operator) sync(ctx context.Context, key string) error {
 	sErr, ok := err.(*apierrors.StatusError)
 
 	if ok && sErr.ErrStatus.Code == 422 && sErr.ErrStatus.Reason == metav1.StatusReasonInvalid {
-		c.metrics.StsDeleteCreateCounter().Inc()
+		if current != nil && current.DeletionTimestamp != nil {
```

Comment:
i think we can check earlier (e.g. before the update) and current should never be nil since we're in the update branch.